src/main/ts/ui/Dialog.ts

```diff
@@ -0,0 +1,137 @@
+    return items.concat({
+      type: 'listbox',
+      name: 'itemsToApplyStyle',
+      label: 'Styles to Applied to',
+      items: [
+        { text: 'The selected list', value: 'selectedList' },
+        { text: 'The selected list + all parent lists', value: 'selectedListAndParent' },
+        { text: 'The selected list + all children lists', value: 'selectedListAndChildren' },
+        { text: 'Current tree', value: 'currentTree' },
+      ]
+    });
+  }
+
+  return items;
+}
+
+const nestedListDialog = (editor: Editor): PublicDialog.DialogSpec<EnhancedListDialogData> => {
+  const selectedElement = SugarElement.fromDom(editor.selection.getNode());
+
+  return {
+    title: 'Enhanced List',
+    body: {
+      type: 'panel',
+      items: getPanelItems(selectedElement),
+    },
+    buttons: [
+      {
+        type: 'cancel',
+        name: 'cancel',
+        text: 'Cancel'
+      },
+      {
+        type: 'submit',
+        name: 'save',
+        text: 'Save',
+        primary: true
+      }
+    ],
+    onSubmit: (api) => {
+      const data = api.getData();
+
+      if (Utils.isListItemNode(selectedElement) || Utils.isListNode(selectedElement)) {
+        const elements = getElementsToApply(selectedElement, data.itemsToApplyStyle);
+        editor.undoManager.transact(() => {
+          Arr.each((elements), (x) => {
+            if ((/^(LI)$/).test(x.nodeName)) {
+              editor.dom.setStyle(x, 'padding-inline-start', `${data.paddingLeft}px`);
+            } else {
+              editor.dom.setStyle(x, 'list-style-type', `${data.listStyle}`);
+            }
+          });
+        });
+      } else {
+        const listStyle = data.listStyle;
+        if (Arr.find(['circle', 'disc', 'square'], (x) => x === listStyle).isSome()) {
+          editor.execCommand('InsertUnorderedList', false, { 'list-style-type': data.listStyle })
+        } else {
+          editor.execCommand('InsertOrderedList', false, { 'list-style-type': data.listStyle })
+        }
+        editor.dom.setStyle(editor.selection.getNode(), 'padding-inline-start', data.paddingLeft + 'px');
+      }
+
+      api.close();
+    }
+  }
+}
+
+const getElementsToApply = (selElement: SugarElement, appliedTo: string): Node[] => {
+  const parent = Traverse.parent(selElement).getOr(selElement);
+  let elements = [parent];
+
+  if (appliedTo === 'selectedListAndParent') {
+    const parents = PredicateFilter.ancestors(parent, Utils.isListNode);
+    elements = elements.concat(parents);
+  } else if (appliedTo === 'selectedListAndChildren') {
+    const children = PredicateFilter.descendants(parent, Utils.isListNode);
+    elements = elements.concat(children);
+  } else if (appliedTo === 'currentTree') {
+    const children = PredicateFilter.descendants(parent, Utils.isListNode);
+    const parents = PredicateFilter.ancestors(parent, Utils.isListNode);
+
+    elements = elements.concat(parents);
+    elements = elements.concat(children);
+  }
```

Comment:
```suggestion

  if (appliedTo === 'selectedListAndParent' || appliedTo === 'currentTree') {
    const parents = PredicateFilter.ancestors(parent, Utils.isListNode);
    elements = elements.concat(parents);
  }
  if (appliedTo === 'selectedListAndChildren' || appliedTo === 'currentTree') {
    const children = PredicateFilter.descendants(parent, Utils.isListNode);
    elements = elements.concat(children);
  }
```
Not much harm in doing it this way, just from what I can see.

Reply:
This is similar to the concept I was talking about above. Except Mitch's example doesn't make the separate object, but just inlines it (effectively). But notice how there is only one call to each of the lookup functions.